### PR TITLE
Add rotation widget to the georeferencer canvas

### PR DESCRIPTION
**Pull request by uclaros**
## Description
You have your maps scanned with arbitrary orientations? No problem!

rotate

### CODE REVIEW

File: src/app/georeferencer/qgsgeorefmainwindow.cpp [hunk 1]
```diff
@@ -1069,6 +1076,26 @@ QLabel *QgsGeoreferencerMainWindow::createBaseLabelStatus()
 
 void QgsGeoreferencerMainWindow::createStatusBar()
 {
+  // add a widget to show/set current rotation
+  mRotationLabel = createBaseLabelStatus();
+  mRotationLabel->setObjectName( QStringLiteral( "mRotationLabel" ) );
+  mRotationLabel->setText( tr( "Rotation" ) );
+  mRotationLabel->setToolTip( tr( "Current clockwise map rotation in degrees" ) );
```
Comment (m-kuhn) [inline src/app/georeferencer/qgsgeorefmainwindow.cpp:1083]:
```suggestion
  mRotationLabel->setToolTip( tr( "Map rotation in degrees clockwise" ) );
```

I think "current" doesn't add a lot and having "map rotation" as first words helps to understand quickly

**Review by m-kuhn — COMMENTED** [review]
Nice.

Did you consider to use a QDial? https://doc.qt.io/qt-5/qdial.html

File: src/app/georeferencer/qgsgeorefmainwindow.cpp (hunk 1 above)
Comment (uclaros) [reply src/app/georeferencer/qgsgeorefmainwindow.cpp:1083]: I kept the tooltip the same as in the main canvas' rotation widget for translation economy. If I indeed change it, the word 'display' could be added so it is clear that the rotation has no effect on georeferencing!

**Comment by uclaros** [comment]
I didn't. My thought was to be similar to the main canvas rotation widget so that users are familiar with its purpose - mostly the fact that it rotates only the display and does not affect georeferencing

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**